src/rewriter/settings.rs

```diff
@@ -5,7 +5,7 @@ use super::AsciiCompatibleEncoding;
 use std::borrow::Cow;
 use std::error::Error;
 
-pub(crate) type HandlerResult = Result<(), Box<dyn Error + Send + Sync>>;
+pub type HandlerResult = Result<(), Box<dyn Error + Send + Sync>>;
```

Comment:
We need doc comments for those and all the newly exposed API. Let's add `#![warn(missing_docs)]` to the crate's lib.rs and also use `RUSTFLAGS="-D warnings" to the CI builds (if we don't have it already)  to fail on missing docs

Reply:
Will add documentation to those fields.  However, we can't fail on missing documentation right because it is missing from a lot of places:

```
warning: `lol_html` (lib) generated 35 warnings
```

Reply:
Let's revisit those and suppress the warning where it's intentional or provide docs otherwise.